src/main/kotlin/no/nav/familie/ks/sak/sikkerhet/TilgangAdvice.kt

```diff
@@ -0,0 +1,183 @@
+import org.aspectj.lang.annotation.Aspect
+import org.aspectj.lang.annotation.Before
+import org.springframework.cache.CacheManager
+import org.springframework.http.HttpMethod
+import org.springframework.stereotype.Component
+import org.springframework.web.context.request.RequestContextHolder
+import org.springframework.web.context.request.ServletRequestAttributes
+import kotlin.reflect.full.declaredMemberProperties
+
+@Aspect
+@Component
+class TilgangAdvice(
+    private val fagsakService: FagsakService,
+    private val behandlingRepository: BehandlingRepository,
+    private val personopplysningGrunnlagRepository: PersonopplysningGrunnlagRepository,
+    private val rolleConfig: RolleConfig,
+    private val integrasjonService: IntegrasjonService,
+    private val cacheManager: CacheManager,
+    private val auditLogger: AuditLogger
+) {
+
+    @Before("@annotation(sjekkTilgang) ")
+    fun sjekkTilgang(joinPoint: JoinPoint, sjekkTilgang: SjekkTilgang) {
+        when (sjekkTilgang.tilgang) {
+            Tilgang.FAGSAK -> {
+                sjekkTilgangTilHandling(sjekkTilgang.minimumBehandlerRolle, sjekkTilgang.handling)
+                val fagsakId =
+                    lesFeltFraKontrollerMetodeParameter(joinPoint, listOf("fagsakId"), sjekkTilgang.index) as Long
+                sjekkTilgangTilFagsak(fagsakId, sjekkTilgang.auditLoggerEvent)
+            }
+
+            Tilgang.BEHANDLING -> {
+                sjekkTilgangTilHandling(sjekkTilgang.minimumBehandlerRolle, sjekkTilgang.handling)
+                val behandlingId =
+                    lesFeltFraKontrollerMetodeParameter(joinPoint, listOf("behandling"), sjekkTilgang.index) as Long
+                sjekkTilgangTilBehandling(behandlingId, sjekkTilgang.auditLoggerEvent)
+            }
+
+            Tilgang.PERSON -> {
+                sjekkTilgangTilHandling(sjekkTilgang.minimumBehandlerRolle, sjekkTilgang.handling)
+                val personIdenter = lesFeltFraKontrollerMetodeParameter(
+                    joinPoint,
+                    listOf("personIdent", "søkersIdent", "ident"),
+                    sjekkTilgang.index
+                ) as String
+                sjekkTilgangTilPersoner(listOf(personIdenter), sjekkTilgang.auditLoggerEvent)
+            }
+
+            Tilgang.HANDLING -> {
+                sjekkTilgangTilHandling(sjekkTilgang.minimumBehandlerRolle, sjekkTilgang.handling)
+            }
+        }
+    }
+
+    private fun lesFeltFraKontrollerMetodeParameter(
+        kontrollerMetode: JoinPoint,
+        feltAliaser: List<String> = emptyList(),
+        index: Int = 0
+    ): Any {
+        val kontrollerMetodeParameter = kontrollerMetode.args[index]
+        if (erGetEllerDelete()) {
+            return kontrollerMetodeParameter
+        }
+        val felter = kontrollerMetodeParameter::class.declaredMemberProperties
+        val felt = felter.first { feltAliaser.contains(it.name) }.getter.call(kontrollerMetodeParameter)
+        return felt!!
```

Comment:
Synes feilmeldingene som `!!` lager er litt diffuse. Kunne vi hatt en egen feil her?
```suggestion
        return felt ?: throw Feil("Klarte ikke å finne felt i request body for ...")
```

Reply:
Jajo. `felt!!` vil kun kaste feil dersom man finner feltet i request-body, men innholdet er `null`. Dersom feltet ikke eksisterer i request-body kastes feilen `NoSuchElementException`, men kan godt wrappe dette i en try catch og kaste `Feil` med en litt bedre feilmelding 👍